src/main/java/com/ms3_inc/camel/extensions/rest/SwaggerRequestValidator.java

```diff
@@ -16,91 +16,143 @@
 
         try {
-            ValidationReport report = validator.validateRequest(fromExchange(exchange));
-            if (report.hasErrors()) {
-                LOGGER.debug(report.toString());
-                exchange.setException(new BadRequestException(fromReport(report)));
-            }
-
-            LOGGER.debug("Validating complete");
-        } catch (Exception ex) {
-            exchange.setException(new CamelException(ex));
+            openapi4jValidator.validate(fromExchange(exchange));
+        } catch (ValidationException e) {
+            ValidationResults results = e.results();
+            LOGGER.debug(e.results().toString());
+            exchange.setException(new BadRequestException(fromReport(results)));
         } finally {
+            LOGGER.debug("Validating complete");
             callback.done(true);
         }
 
         return true;
     }
 
-    private static OperationResult fromReport(ValidationReport report) {
+    private static OperationResult fromReport(ValidationResults results) {
         List<OperationResult.Message> answer = new ArrayList<>(1);
+        StringBuilder resultItems = new StringBuilder();
 
-        // separates the report into a "Validation failed." line for details and the rest for diagnostics
-        String[] result = SimpleValidationReportFormat.getInstance().apply(report).split(System.lineSeparator(), 2);
-        answer.add(new OperationResult.Message(OperationResult.Level.ERROR, null, result[0], result[1]));
+        // separates the results into a "Validation failed." line for details and the rest for diagnostics
+        for (ValidationResults.ValidationItem result : results.items()) {
+            resultItems.append(result.message() + "\n");
+        }
+        answer.add(new OperationResult.Message(OperationResult.Level.ERROR, null, "Validation failed.", resultItems.toString()));
 
         return new OperationResult(answer);
     }
 
     private static Request fromExchange(Exchange exchange) {
+        final Logger LOGGER = LoggerFactory.getLogger(SwaggerRequestValidator.class);
+
         final Request.Method method = Request.Method.valueOf(exchange.getMessage().getHeader(Exchange.HTTP_METHOD, String.class));
         final String path = exchange.getMessage().getHeader(Exchange.HTTP_URI, String.class);
         final String query = exchange.getMessage().getHeader(Exchange.HTTP_QUERY, String.class);
         final String body = MessageHelper.extractBodyAsString(exchange.getMessage());
-        final SimpleRequest.Builder requestBuilder = new SimpleRequest.Builder(method, path);
+        final String contentType = exchange.getMessage().getHeader(Exchange.CONTENT_TYPE, String.class);
+        final DefaultRequest.Builder requestBuilder = new DefaultRequest.Builder(path, method);
 
         if (!body.isEmpty()) {
-            requestBuilder.withBody(body);
+            if (contentType.equals("application/xml")) {
+                XmlMapper xmlMapper = new XmlMapper();
```

Comment:
Let's make this a class constant, so we don't have to instantiate with every request

Reply:
The XmlMapper specifically, like  `final XmlMapper xmlMapper = new XmlMapper();` ?

Reply:
yeah private final static for the class